library/src/main/java/com/mapbox/android/gestures/MoveGestureDetector.java

```diff
@@ -172,10 +178,11 @@ private void updateMoveDistancesObjects() {
 
   boolean checkAnyMoveAboveThreshold() {
     for (MoveDistancesObject moveDistancesObject : moveDistancesObjectMap.values()) {
-      if (Math.abs(moveDistancesObject.getDistanceXSinceStart()) >= moveThreshold
-        || Math.abs(moveDistancesObject.getDistanceYSinceStart()) >= moveThreshold) {
-        return true;
-      }
+      boolean thresholdExceeded = Math.abs(moveDistancesObject.getDistanceXSinceStart()) >= moveThreshold
+        || Math.abs(moveDistancesObject.getDistanceYSinceStart()) >= moveThreshold;
+
+      boolean isInRect = moveThresholdRect != null && moveThresholdRect.contains(getFocalPoint().x, getFocalPoint().y);
+      return !isInRect && thresholdExceeded;
```

Comment:
nit: if the focal point is not in rect, we don't have to check the threshold any more? I don't know if the check is computationally heavy but I assume this method is called frequently.

Reply:
> if the focal point is not in rect, we don't have to check the threshold any more?

we are checking that - if the point is not in rect, it also has to exceed the threshold for `checkAnyMoveAboveThreshold` to return `true`.

> if the check is computationally heavy

doesn't look like it, also it's not checked after the gesture is started anymore

Reply:
> we are checking that

Added a test to confirm https://github.com/mapbox/mapbox-gestures-android/pull/96/commits/0f23bb14e290504bdb6cf009e89e75e7afe3bb05.